index.d.ts

```diff
@@ -1,14 +1,16 @@
 export class AbortError extends Error {
+	readonly name: 'AbortError';
```

Comment:
Why is this needed? The name is already defined in the `class AbortError`.

Reply:
It can be useful if you try to find out the type of the error without using `typeof`:

```ts
class E extends Error {
    name: 'E';
    foo: 'bar';
}

const e = new E();

if (e.name === 'E') {
    e.foo; // <- no error here
}
```

Reply:
I suppose for the sake of completeness and for the case that you use this error in a discriminated union type:

```ts
class E extends Error {
    name: 'E';
    foo: 'bar';
}

class F extends Error {
    name: 'F';
    baz: 'bar';
}

var e : E | F = new E();

if (e.name === 'E') {
    e.foo; // <- no TS error here
}
```

Reply:
Makes sense, but I'm surprised TS does not make the `.name` property the same as the name of the class...

Reply:
Well, this is not the class property that is being set here (`AbortError.name`) but the instance property name (`err.name`). The first is being set automatically, the second not.